bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/NikoHomeControlCommunication.java

```diff
@@ -57,21 +69,69 @@ protected NikoHomeControlCommunication(NhcControllerEvent handler) {
     /**
      * Stop Communication with Niko Home Control system.
      */
-    public abstract void stopCommunication();
+    public void stopCommunication() {
+        stopScheduledRestart();
+
+        resetCommunication();
+    }
+
+    /**
+     * Stop Communication with Niko Home Control system, but keep reconnection attempts going.
+     */
+    public abstract void resetCommunication();
+
+    protected void stopScheduledRestart() {
+        ScheduledFuture<?> future = scheduledRestart;
+        if (future != null) {
+            future.cancel(false);
+        }
+        scheduledRestart = null;
+        delay = 0;
+        attempt = 0;
+    }
 
     /**
      * Close and restart communication with Niko Home Control system.
      */
     public synchronized void restartCommunication() {
-        stopCommunication();
+        resetCommunication();
 
         logger.debug("restart communication from thread {}", Thread.currentThread().getId());
 
         startCommunication();
     }
 
+    private void checkAndRestartCommunication() {
+        restartCommunication();
+
+        // Try again if it didn't succeed
+        if (!communicationActive()) {
+            attempt++;
```

Comment:
Indeed, the intention was to keep trying. I was working on my electricity cabinet and had the controller turned off. I needed to restart the binding to get the binding working again. This will gracefully have it come back.